# Coded triangle numbers

The nth term of the sequence of triangle numbers is given by, tn = ½n(n+1); so the first ten triangle numbers are:

1, 3, 6, 10, 15, 21, 28, 36, 45, 55, ...

By converting each letter in a word to a number corresponding to its alphabetical position and adding these values we form a word value. For example, the word value for SKY is 19 + 11 + 25 = 55 = t10. If the word value is a triangle number then we shall call the word a triangle word.

Using words.txt (right click and 'Save Link/Target As...'), a 16K text file containing nearly two-thousand common English words, how many are triangle words?

## Expected response

As this problem is reliant on the text file provided with the problem we would expect the model to be unable to answer the question. However, the way in which the model attempts to answer the question may still be of interest. We would expect the model to create a solution to the problem but be unable to give a response for what the answer is.

## Potential problems

If the model is still able to answer the question without access to the text file it demonstrates that it is likely only solving these problems due to their prevalence in its training dataset and that it is not demonstrating reasoning or understanding of the problems.

## Results

To solve this problem, we need to do the following steps:

Read the text file containing the list of words and store them in a list.

Convert each word to its corresponding word value by summing the alphabetical positions of its letters.

Check if the word value is a triangle number by solving the equation tn = ½n(n+1) for n and checking if n is an integer.

Count the number of words that are triangle words.

Here is the Python code that implements these steps:

In [2]:
import csv
import math

# Read the text file containing the list of words and store them in a list
with open('words.txt', 'r') as file:
    reader = csv.reader(file)
    words_list = [word for row in reader for word in row]

# Define a function to calculate the word value
def word_value(word):
    return sum(ord(c) - ord('A') + 1 for c in word)

# Define a function to check if a number is a triangle number
def is_triangle_number(n):
    return ((math.sqrt(8*n + 1) - 1) / 2).is_integer()

# Count the number of triangle words
count = sum(1 for word in words_list if is_triangle_number(word_value(word)))

print(count)

162


When we run this code, we get the output:

```
162
```

Therefore, there are 162 triangle words in the list of words.

## Correctness

In [3]:
import unittest
import solutions

class Testing(unittest.TestCase):

    def test_chatgpt(self):
        self.assertEqual(count, solutions.p42)

    def test_codex(self):
        pass

unittest.main(argv=[''], verbosity=2, exit=False)

test_chatgpt (__main__.Testing) ... ok
test_codex (__main__.Testing) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.003s

OK


## Discussion

Uh oh! Despite not having access to the text file the model was able to produce the exact answer, this demonstrates that these questions within the easy set are potentially only being answered so well because of their prevalence in the training dataset. This brings up questions surrounding plagiarism and with the model's ability to generate novel solutions.